src/super_gradients/common/deprecate.py

```diff
@@ -0,0 +1,79 @@
+import warnings
+from functools import wraps
+from typing import Optional
+
+
+def deprecate_call(deprecated_in_v: str, removed_in_v: str, target: Optional[callable] = None, reason: str = ""):
```

Comment:
why not "deprecated"?
not removed_in_v, remove_in_v

Reply:
This was designed for **callable**. We may want to add a deprecated decorator for **parameters** (that's a common use-case as well)
So I thought it would be better to distinguish between both with a clear name. But I'm also ok with `deprecate`, we can go for it if you still think it's better